### ISSUE TITLE
Testing the latest DAGMC with latest OpenMC

### ISSUE THREAD

**Issue by shimwell** [svalinn/DAGMC#795]
**Is your feature request related to an enhancement? Please describe.**<br/>
We could add some CI to DAGMC that checks that the compiling of DAGMC with OpenMC continues to work

**Is your feature request related to a problem? Please describe.**<br/>
It is currently possible that we merge code into DAGMC that for some reason does not compile with OpenMC.
I believe this is performed manually currently.

**Describe the solution or outcome you'd like.**<br/>
A new GitHub action that downloads the development branch of OpenMC and compiles OpenMC with DAGMC enabled.

**Describe alternatives you've considered.**<br/>
Ideally some DAGMC specific OpenMC tests would be run to check that the combination of the codes work as expected. However I think that can be a follow up feature PR.
This has not been possible to carry out in the public CI with other export controlled codes, so I don't think we can consider doing this with other neutronics codes.

**Comment by pshriwise** [comment]
What would you think of doing this only on pushes to `develop`?

**Comment by makeclean** [comment]
I wonder if this shouldn't be the other way around? We build and test the dagmc codes, as these are 'our' dagmc versions of those codes, fludag and daggeant4 etc i.e. we depend upon those codes. Where as dagmc is a dependency of openmc, I think openmc should check to see if it builds against dagmc.

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by pshriwise** [comment]
Yeah, you're right @makeclean. I'm having deja vu as I'm pretty sure we've had this whole conversation before and somehow I've totally forgotten 😆 

@shimwell we test against DAGMC's develop branch in OpenMC's CI. This is only checked when OpenMC PR's are merged of course, not when new changes come into DAGMC.

If we'd like to test more frequently, maybe a separate repo specifically for this purpose with a GHA triggered by either a merged OpenMC or DAGMC PR would be the way to go.

**Comment by shimwell** [comment]
> If we'd like to test more frequently, maybe a separate repo specifically for this purpose with a GHA triggered by either a merged OpenMC or DAGMC PR would be the way to go.

That sounds interesting, I am just not sure how to do it
